### DISCUSSION TITLE
v5.0.0

### DISCUSSION THREAD

**Discussion by richardwilkes** [richardwilkes/gcs#533] [Announcements]
### New Features
- Better performance in many areas
- Integrated viewers for PDF, PNG, JPEG, WEBP, GIF, BMP, WBMP, ICO, and a limited form of Markdown
- Most modal dialogs have been eliminated, replaced with editors that can be managed like any other tabbed view in the workspace
- A large number of minor and sometimes not-so-minor enhancements that I've unfortunately failed to record over the past year...

### Library Compatibility
- Loads library file formats v3.x.x and v4.x.x. Only produces v4.x.x. Note that GCS v4.30.0 is the oldest version that produced the v3.x.x library file format. You will need to load and save your data with a version capable of writing the v3.x.x format before you will be able to load it into this release.

<hr /><em>This discussion was created from the release <a href='https://github.com/richardwilkes/gcs/releases/tag/v5.0.0'>v5.0.0</a>.</em>

**Comment by Delnorin** [comment]
I installed 5.0.0
I made some adjustments to my character and then saved the file.
I then went to export and it was no longer PNG, but PDF.  Fine, okay, doesn't matter.  But then it just locked up and the program failed to work.
Also, there is no longer an export to the GURP website.

Bah..

So I went back to version 4.37.1
It won't load my character now.. version 5.0.0 changed its format and so now I have to rebuild the character completely.

Okay, annoying but only 2 hours out of my life to fix it.

Rebuilding character now...
But...
Libraries were updated with 5.0.0 and no longer work and can not erase them/update them.

Ah, now version 5.0.1 came out.
But my libraries don't work now.

Uninstall...
Reinstall...
No libraries.

Tonight's simple (update button) has turned into an all night nightmare.

**Comment by richardwilkes** [comment]
Sorry to hear you (and several others) ran afoul of a bug in the code that loaded settings -- and blew up when it encountered certain older settings. v5.0.1 should fix that. If you aren't seeing the master library content, you just need to have GCS re-download it. There is a menu choice for that in v4.x.x... or a button for it in v5.

**Reply by Delnorin** [reply]
Thank you Sir.
I have wiped everything.
Installed v5.0.1
I have pushed the magic button and it has reloaded my libraries.

Do you plan on offering the Export to gurpscalculator.com with this program any longer?

Also, when I Export -> PDF
It is still freezing the program completely and requiring a File Explorer -end program-... even after 30 minutes of trying to make the pdf file.

update...
Also if I select -Print- it freezes the program.

**Reply by richardwilkes** [reply]
gcalc support has been dropped. The site does not seem to be actively maintained and when I did a poll to see what percentage of users actually made use of it, the result was close to zero. I will probably incorporate some of the helpers it had at some point.

As for export to PDF failing… please create an issue for it, including what OS you’re using and the exact steps you’re taking. If you can attach a file that triggers the problem, even better.

Same request for printing. I will note that the first thing printing does is try to find your printer via IPP (Internet Printing Protocol). Most printers in the past 15-20 years support that, but not all. If you don’t have a printer capable of IPP, that might be the problem. The UI shouldn’t freeze, but I may not have actually tested that case, unfortunately. Again, creating a bug report with exact steps taken will be helpful.

**Reply by richardwilkes** [reply]
Actually, now that I think about it some more, the printing issue is likely just due to the PDF generation failing, since a PDF is generated to send to the printer.

**Reply by Delnorin** [reply]
I am going to post a video on Rumble and send you the link.  I tried to attach it and the video file is too long (600 MB).  It's a "bug report" that is hopefully helpful.

Sorry it took so long to respond.  I was not at home previously and wanted to make sure I was home and attempted the process at home as well to rule out any 'network permissions/printer' issues that may have been occurring where I was logged in from earlier.

Attempted to attach the *.gcs file to this but the website does not accept that file format.

[Video Here](https://rumble.com/v1lok87-gcs-bug-report-9-26-2022.html)
Rumble is decoding the video so if you only get a lame 480p video quality... just come back later.. it's loaded as an amazing 1440p once Rumble is done processing it.

**Reply by richardwilkes** [reply]
Thanks. Please send me the .gcs file that is causing the problem. I noticed you have a very large note in that file. I suspect the algorithm I wrote to break the content up into pages for PDF output is getting stuck because it is trying to fit that single note onto a page and can't. I'll work on a fix. In the meantime, if you break that note up into smaller chunks by hand, I suspect it will allow you to generate a PDF and print.

**Reply by richardwilkes** [reply]
In case I wasn't being precise enough, I think it was the "Monica Zeller.gcs" file.

**Reply by richardwilkes** [reply]
If breaking the note up doesn't get you unstuck, you could also try exporting it to the "HTML - PC" template, which will emit an html file for you.

**Reply by Delnorin** [reply]
That was the trick.  I brought her Background Note into 4 subsections and the pdf was created perfectly.
So stoked.  Thank you so much Richard!

**Comment by Delnorin** [comment]
Was the Basic Set Skill "Electronics Operation/TL (Navigation) Removed?
Or was that a specialized @Electronics Type@ that I made on my own in the past and the new/refreshed libraries do not have that?

**Comment by richardwilkes** [comment]
I'm not aware of any recent changes in that area. Certainly no changes due to v5 yet, other than a mass conversion to the new format, which didn't remove anything.

**Reply by Delnorin** [reply]
Good deal then.  It must have been a @Specialization@ that I had made earlier and modified my earlier library.
Thank you Sir.
